File: add-duty-menu.js
// 用于添加值班管理菜单的脚本
// 可以通过前端控制台或Node.js运行此脚本

// 菜单数据
const menuData = [
  // 值班管理目录菜单
  {
    menuName: '值班管理',
    parentId: 0, // 顶级菜单
    path: '/duty',
    component: 'views/duty/DutyLayout.vue',
    perm: '',
    type: 1, // 1=目录
    icon: 'Calendar',
    sort: 3, // 排在系统管理后面
    status: 1
  },
  // 值班表管理菜单
  {
    menuName: '值班表管理',
    parentId: null, // 会在添加父菜单后更新
    path: '/duty/schedule',
    component: 'views/duty/DutySchedule.vue',
    perm: 'duty:schedule:list',
    type: 2, // 2=菜单
    icon: 'DocumentCopy',
    sort: 1,
    status: 1
  },
  // 值班安排菜单
  {
    menuName: '值班安排',
    parentId: null, // 会在添加父菜单后更新
    path: '/duty/assignment',
    component: 'views/duty/DutyAssignment.vue',
    perm: 'duty:assignment:list',
    type: 2, // 2=菜单
    icon: 'CalendarCheck',
    sort: 2,
    status: 1
  },
  // 值班记录菜单
  {
    menuName: '值班记录',
    parentId: null, // 会在添加父菜单后更新
    path: '/duty/record',
    component: 'views/duty/DutyRecord.vue',
    perm: 'duty:record:list',
    type: 2, // 2=菜单
    icon: 'Document',
    sort: 3,
    status: 1
  }
];

// 模拟API请求函数
const request = async (url, method, data) => {
  const response = await fetch(`http://localhost:8080/api${url}`, {
    method: method,
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${localStorage.getItem('token')}`
    },
    body: data ? JSON.stringify(data) : null
  });
  return response.json();
};

// 添加菜单的函数
const addMenu = async (menu) => {
  return request('/menu', 'POST', menu);
};

// 执行添加菜单的逻辑
const addDutyMenu = async () => {
  try {
    // 1. 添加值班管理目录菜单
    console.log('添加值班管理目录菜单...');
    const dutyMenuResponse = await addMenu(menuData[0]);
    const dutyMenuId = dutyMenuResponse.data.id;
    console.log('值班管理目录菜单添加成功，ID:', dutyMenuId);
    
    // 2. 更新子菜单的parentId
    menuData[1].parentId = dutyMenuId;
    menuData[2].parentId = dutyMenuId;
    menuData[3].parentId = dutyMenuId;
    
    // 3. 添加子菜单
    for (let i = 1; i < menuData.length; i++) {
      const subMenu = menuData[i];
      console.log(`添加${subMenu.menuName}菜单...`);
      const subMenuResponse = await addMenu(subMenu);
      console.log(`${subMenu.menuName}菜单添加成功，ID:`, subMenuResponse.data.id);
    }
    
    console.log('所有值班管理菜单添加成功！');
  } catch (error) {
    console.error('添加菜单失败:', error);
  }
};

// 如果在浏览器环境中运行
if (typeof window !== 'undefined') {
  // 将函数暴露给全局，以便在控制台调用
  window.addDutyMenu = addDutyMenu;
  console.log('添加值班管理菜单的函数已就绪，请调用 addDutyMenu() 执行添加操作。');
} else {
  // 如果在Node.js环境中运行，直接执行
  addDutyMenu();
}

File: AddDutyMenu.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

/**
 * 用于添加值班管理菜单的Java程序
 */
public class AddDutyMenu {
    
    // 数据库连接信息
    private static final String URL = "jdbc:mysql://localhost:3306/hyper_duty?useUnicode=true&zeroDateTimeBehavior=convertToNull&useSSL=true&serverTimezone=GMT%2B8&useLegacyDatetimeCode=false";
    private static final String USER = "root";
    private static final String PASSWORD = "Feiwu2023!";
    
    public static void main(String[] args) {
        Connection connection = null;
        Statement statement = null;
        
        try {
            // 加载驱动
            Class.forName("com.mysql.cj.jdbc.Driver");
            
            // 建立连接
            connection = DriverManager.getConnection(URL, USER, PASSWORD);
            statement = connection.createStatement();
            
            System.out.println("连接数据库成功！");
            
            // 1. 添加值班管理目录菜单
            String addDutyMenu = "INSERT INTO sys_menu (menu_name, parent_id, path, component, perm, type, icon, sort, status, create_time, update_time) "
                              + "VALUES ('值班管理', 0, '/duty', 'views/duty/DutyLayout.vue', '', 1, 'Calendar', 3, 1, NOW(), NOW())";
            
            statement.executeUpdate(addDutyMenu, Statement.RETURN_GENERATED_KEYS);
            
            // 获取刚插入的值班管理目录ID
            long dutyMenuId = 0;
            ResultSet generatedKeys = statement.getGeneratedKeys();
            if (generatedKeys.next()) {
                dutyMenuId = generatedKeys.getLong(1);
            }
            
            System.out.println("添加值班管理目录菜单成功，ID: " + dutyMenuId);
            
            // 2. 添加值班表管理菜单
            String addScheduleMenu = "INSERT INTO sys_menu (menu_name, parent_id, path, component, perm, type, icon, sort, status, create_time, update_time) "
                                  + "VALUES ('值班表管理', ?, '/duty/schedule', 'views/duty/DutySchedule.vue', 'duty:schedule:list', 2, 'DocumentCopy', 1, 1, NOW(), NOW())";
            
            PreparedStatement scheduleStmt = connection.prepareStatement(addScheduleMenu);
            scheduleStmt.setLong(1, dutyMenuId);
            scheduleStmt.executeUpdate();
            System.out.println("添加值班表管理菜单成功");
            
            // 3. 添加值班安排菜单
            String addAssignmentMenu = "INSERT INTO sys_menu (menu_name, parent_id, path, component, perm, type, icon, sort, status, create_time, update_time) "
                                    + "VALUES ('值班安排', ?, '/duty/assignment', 'views/duty/DutyAssignment.vue', 'duty:assignment:list', 2, 'Calendar', 2, 1, NOW(), NOW())";
            
            PreparedStatement assignmentStmt = connection.prepareStatement(addAssignmentMenu);
            assignmentStmt.setLong(1, dutyMenuId);
            assignmentStmt.executeUpdate();
            System.out.println("添加值班安排菜单成功");
            
            // 4. 添加值班记录菜单
            String addRecordMenu = "INSERT INTO sys_menu (menu_name, parent_id, path, component, perm, type, icon, sort, status, create_time, update_time) "
                                + "VALUES ('值班记录', ?, '/duty/record', 'views/duty/DutyRecord.vue', 'duty:record:list', 2, 'Document', 3, 1, NOW(), NOW())";
            
            PreparedStatement recordStmt = connection.prepareStatement(addRecordMenu);
            recordStmt.setLong(1, dutyMenuId);
            recordStmt.executeUpdate();
            System.out.println("添加值班记录菜单成功");
            
            // 5. 查询添加的菜单，验证结果
            String queryMenus = "SELECT * FROM sys_menu WHERE parent_id = " + dutyMenuId + " OR id = " + dutyMenuId;
            ResultSet resultSet = statement.executeQuery(queryMenus);
            
            System.out.println("\n添加的菜单列表：");
            while (resultSet.next()) {
                long id = resultSet.getLong("id");
                String menuName = resultSet.getString("menu_name");
                long parentId = resultSet.getLong("parent_id");
                String path = resultSet.getString("path");
                String component = resultSet.getString("component");
                int type = resultSet.getInt("type");
                String icon = resultSet.getString("icon");
                int status = resultSet.getInt("status");
                
                System.out.println("ID: " + id + ", 菜单名称: " + menuName + ", 父ID: " + parentId + ", 路径: " + path + ", 组件: " + component + ", 类型: " + type + ", 图标: " + icon + ", 状态: " + status);
            }
            
            System.out.println("\n所有值班管理菜单添加成功！");
            
        } catch (ClassNotFoundException e) {
            System.out.println("找不到MySQL驱动类：" + e.getMessage());
        } catch (SQLException e) {
            System.out.println("数据库操作失败：" + e.getMessage());
            e.printStackTrace();
        } finally {
            // 关闭资源
            try {
                if (statement != null) {
                    statement.close();
                }
                if (connection != null) {
                    connection.close();
                }
            } catch (SQLException e) {
                System.out.println("关闭资源失败：" + e.getMessage());
            }
        }
    }
}
